ballerina-tests/tests/37_interfaces.bal

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/test;
+
+@test:Config {
+    groups: ["interfaces"]
+}
+isolated function testInterfaces() returns error? {
+    string document = check getGraphQLDocumentFromFile("interfaces.graphql");
+    string url = "http://localhost:9098/interfaces";
+    json actualPayload = check getJsonPayloadFromService(url, document);
+    json expectedPayload = check getJsonContentFromFile("interfaces.json");
+    assertJsonValuesWithOrder(actualPayload, expectedPayload);
+}
+
+@test:Config {
+    groups: ["interfaces", "introspection"]
+}
+isolated function testInterfaceIntrospection() returns error? {
+    string document = check getGraphQLDocumentFromFile("interface_introspection.graphql");
+    string url = "http://localhost:9098/interfaces";
+    json actualPayload = check getJsonPayloadFromService(url, document);
+    json expectedPayload = check getJsonContentFromFile("interface_introspection.json");
+    assertJsonValuesWithOrder(actualPayload, expectedPayload);
+}
+
+@test:Config {
+    groups: ["interfaces"]
+}
+isolated function testInterfaceField() returns error? {
+    string document = check getGraphQLDocumentFromFile("interface_field.graphql");
+    string url = "http://localhost:9098/interfaces";
+    json actualPayload = check getJsonPayloadFromService(url, document);
+    json expectedPayload = check getJsonContentFromFile("interface_field.json");
+    assertJsonValuesWithOrder(actualPayload, expectedPayload);
+}
```

Comment:
Does the compiler plugin test cover all the negative test cases related to `interfaces`? Don't we need to add negative test cases here?

Reply:
What kind of negative tests are we looking at here? Compiler plugin tests are checking for invalid interface definitions. (We cannot test them in Ballerina tests, right).